Sources/PackageCollectionGenerator/PackageMetadataProviders/GitLabPackageManagerMetadataProvider.swift

```diff
@@ -0,0 +1,170 @@
+        self.httpClient = httpClient ?? Self.makeDefaultHTTPClient()
+        self.decoder = JSONDecoder.makeWithDefaults()
+    }
+
+    func get(_ packageURL: URL, callback: @escaping (Result<PackageBasicMetadata, Error>) -> Void) {
+        guard let baseURL = self.apiURL(packageURL.absoluteString),
+              let urlComponents = URLComponents(url: baseURL, resolvingAgainstBaseURL: false) else {
+                  return callback(.failure(Errors.invalidGitURL(packageURL)))
+              }
+
+        let projectPath = urlComponents.path.dropFirst().replacingOccurrences(of: "/", with: "%2F")
+        let apiPrefix = GitLabPackageMetadataProvider.apiHostURLPathPostfix
+        let metadataURL = URL(string: "\(urlComponents.scheme!)://\(urlComponents.host!)/\(apiPrefix)/projects/\(projectPath)")!
+
+        // get the main data
+        let metadataHeaders = HTTPClientHeaders()
+        let metadataOptions = self.makeRequestOptions(validResponseCodes: [200, 401, 403, 404])
+        let hasAuthorization = metadataOptions.authorizationProvider?(metadataURL) != nil
+        var result: Result<HTTPClient.Response, Error> = tsc_await { callback in self.httpClient.get(metadataURL, headers: metadataHeaders, options: metadataOptions, completion: callback) }
+
+        if case .success(let response) = result {
+            let apiLimit = response.headers.get("RateLimit-Limit").first.flatMap(Int.init) ?? -1
+            let apiRemaining = response.headers.get("RateLimit-Remaining").first.flatMap(Int.init) ?? -1
+            switch (response.statusCode, hasAuthorization, apiRemaining) {
+            case (_, _, 0):
+                result = .failure(Errors.apiLimitsExceeded(metadataURL, apiLimit, apiRemaining))
+            case (401, true, _):
+                result = .failure(Errors.invalidAuthToken(metadataURL))
+            case (401, false, _):
+                result = .failure(Errors.permissionDenied(metadataURL))
+            case (403, _, _):
+                result = .failure(Errors.permissionDenied(metadataURL))
+            case (404, _, _):
+                result = .failure(Errors.notFound(metadataURL))
+            case (200, _, _):
+                guard let metadata = try? response.decodeBody(GetRepositoryResponse.self, using: self.decoder) else {
+                    callback(.failure(Errors.invalidResponse(metadataURL, "Invalid body")))
+                    return
+                }
+
+                let license = metadata.license
+                let packageLicense: PackageCollectionModel.V1.License?
+                if let license = license {
+                    packageLicense = .init(name: license.name, url: license.sourceURL)
+                } else if let licenseURL = metadata.license_url,
+                    let licenseURL = URL(string: licenseURL) {
+                    packageLicense = .init(name: nil, url: licenseURL)
+                } else {
+                    packageLicense = nil
+                }
```

Comment:
Judging from the [sample response](https://docs.gitlab.com/ee/api/projects.html#get-single-project):

```
  "license_url": "http://example.com/diaspora/diaspora-client/blob/master/LICENSE",
  "license": {
    "key": "lgpl-3.0",
    "name": "GNU Lesser General Public License v3.0",
    "nickname": "GNU LGPLv3",
    "html_url": "http://choosealicense.com/licenses/lgpl-3.0/",
    "source_url": "http://www.gnu.org/licenses/lgpl-3.0.txt"
  },
```

`packageLicense` should have `name` set to `license.key` and `url` to `metadata.licenseURL`.